commerce_coordinator/apps/titan/pipeline.py

```diff
@@ -140,3 +144,25 @@ def run_filter(self, edx_lms_user_id):  # pylint: disable=arguments-differ
             'order_data': order_data,
             'recent_payment': recent_payment
         }
+
+
+class UpdateBillingAddress(PipelineStep):
+    """
+    Updates the user's billing address info on an order in Titan
+    """
+
+    def run_filter(self, order_uuid, **kwargs):  # pylint: disable=arguments-differ
+
+        api_client = TitanAPIClient()
+        try:
+            response = api_client.update_billing_address(order_uuid, **kwargs)
+
+        except HTTPError as exc:
+            logger.exception(
+                "[UpdateBillingAddress] Failed to update the billing address for the specified order: %s",
+                order_uuid
+            )
+            raise APIException("Error updating the order's billing address details in titan") from exc
```

Comment:
@wdrussell2015, could you file a ticket to incorporate error handling for titan/pipeline.py? There are a lot of errors from the API spec that we collectively as a team haven't implemented in this file.

For example, I don't think right now we would know the difference between, e.g. an `orderHasCompletedPayments` and an `orderNotFound` error on this endpoint (see [api spec](https://github.com/edx/commerce-coordinator-private/blob/a740ebf00ffee93efac0a61c7ed47293f21511b6/docs/api-spec/theseus-api-spec.md#action-update-billing-address)). We should make sure this occurrence makes it into the logs, at the very least.

(I was going to suggest adding this exception into exception.py to follow suit from some of the other pipeline steps. But it looks like you don't have enough information from the client in this scope to be able to name what exception is happening properly. This work will also have to be done in the ticket above.)

Reply:
https://2u-internal.atlassian.net/browse/THES-223